lib/beaker/dsl/install_utils.rb

```diff
@@ -807,8 +807,8 @@ def install_puppet_from_msi( host, opts )
         end
 
         if host['is_cygwin'].nil? or host['is_cygwin'] == true
-          dest = "/cygdrive/c/Windows/Temp/#{host['dist']}.msi"
-          on host, "curl -O #{dest} #{link}"
+          dest = "#{host['dist']}.msi"
+          on host, "curl -O #{link}"
```

Comment:
msiexec can install MSI's from a URL, so we don't actually need to curl it down.

Also `msiexec` will install puppet asynchronously. To block waiting for the install to complete you have to do:

`"cmd /c start /w msiexec /qn /i #{link}"`